### PR TITLE
Fix + refactor setup.py

### PR DESCRIPTION
**Pull request by TheTripleV**

### CODE REVIEW

**Comment by TheTripleV** [comment]
From commit message:
- Switch file ops to pathlib
- fix: requires_python -> python_requires
- fix: remove newlines, blanklines, comments from install_requires and extras_require
- fix: dont use absolute path for req keys

File: setup.py [hunk 1]
```diff
@@ -41,32 +39,37 @@
...
 
-all_reqs = set()
 
-with open(join(setup_dir, "requirements.txt")) as requirements_file:
-    install_requires = requirements_file.readlines()
-    all_reqs |= set(install_requires)
+def get_reqs_from_path(path):
+    return [
+        req.strip()
+        for req in Path(path).read_text().splitlines()
```
Comment (auscompgeek) [inline setup.py:52]: Hmm. `readlines` would definitely be better here.

Comment (TheTripleV) [reply setup.py:52]: Path doesn't have a `readlines`.

Comment (auscompgeek) [reply setup.py:52]: A file has a `readlines` though, so I don't see your point.

Comment (TheTripleV) [reply setup.py:52]: I don't think it makes a big difference either way.

```py
with open(path, "r") as f:
	lines = f.readlines()

return [
      req.strip()
      for req in lines
...
```

vs 
```py
return [
    req.strip()
    for req in Path(path).read_text().splitlines()
...
```

**Pull request merged**